manager/scheduler/nodeinfo.go

```diff
@@ -1,13 +1,17 @@
 package scheduler
 
 import (
+	"fmt"
+	"strings"
 	"time"
 
 	"github.com/docker/swarmkit/api"
 	"github.com/docker/swarmkit/log"
 	"golang.org/x/net/context"
 )
 
+type tprTaskStore map[string]map[string]*api.Set
```

Comment:
Could this be `map[string]map[string]map[string]struct{}`? That would avoid some of the awkwardness with empty `&api.Set_Void{}` values.